compiler/modules/front/codeGen.bal

```diff
@@ -648,7 +671,7 @@ function maybeCreateBasicBlock(CodeGenContext cx, bir:BasicBlock? block) returns
 
 function codeGenIfElseStmt(CodeGenContext cx, bir:BasicBlock startBlock, Environment env, s:IfElseStmt stmt) returns CodeGenError|StmtEffect {
     var { condition, ifTrue, ifFalse } = stmt;
-    var { result: operand, block: branchBlock, narrowing } = check codeGenConditionalExpr(cx, startBlock, env, condition);
+    var { result: operand, block: branchBlock, narrowing: exprNarrowing } = check codeGenConditionalExpr(cx, startBlock, env, condition);
```

Comment:
I suggest condNarrowing rather than exprNarrowing (since you are using if/elseNarrowings not stmtNarrowings below).